numba_cuda/numba/cuda/tests/nrt/test_nrt.py

```diff
@@ -0,0 +1,108 @@
+import gc
+import numpy as np
+import unittest
+from unittest.mock import patch
+from numba.core.runtime import rtsys
+from numba.tests.support import TestCase, EnableNRTStatsMixin
+from numba.cuda.testing import CUDATestCase
+
+from numba import cuda
+
+from numba.core import errors, types
+from numba.core.extending import overload
+from numba.np.arrayobj import (_check_const_str_dtype, is_nonelike,
+                               ty_parse_dtype, ty_parse_shape, numpy_empty_nd)
+
+
+def cuda_empty(shape, dtype):
+    pass
+
+
+@overload(cuda_empty)
+def ol_cuda_empty(shape, dtype):
+    _check_const_str_dtype("empty", dtype)
+    if (dtype is float or
+        (isinstance(dtype, types.Function) and dtype.typing_key is float) or
+            is_nonelike(dtype)): #default
+        nb_dtype = types.double
+    else:
+        nb_dtype = ty_parse_dtype(dtype)
+
+    ndim = ty_parse_shape(shape)
+    if nb_dtype is not None and ndim is not None:
+        retty = types.Array(dtype=nb_dtype, ndim=ndim, layout='C')
+
+        def impl(shape, dtype):
+            return numpy_empty_nd(shape, dtype, retty)
+        return impl
+    else:
+        msg = f"Cannot parse input types to function np.empty({shape}, {dtype})"
+        raise errors.TypingError(msg)
+
+
+@unittest.skip
+class TestNrtRefCt(EnableNRTStatsMixin, TestCase):
```

Comment:
Should this be

```suggestion
class TestNrtRefCt(EnableNRTStatsMixin, CUDATestCase):
```

?

Reply:
I know it's skipped, but oddly it seems not to even get discovered when I'm running the tests locally:

```
$ python -m numba.runtests numba.cuda.tests.nrt.test_nrt -v
test_nrt_launches (numba.cuda.tests.nrt.test_nrt.TestNrtBasic.test_nrt_launches) ... ok
test_nrt_returns_correct (numba.cuda.tests.nrt.test_nrt.TestNrtBasic.test_nrt_returns_correct) ... ok

----------------------------------------------------------------------
Ran 2 tests in 2.926s

OK
```

Reply:
Maybe this is something odd about `unittest`. If I change it to:

```diff
diff --git a/numba_cuda/numba/cuda/tests/nrt/test_nrt.py b/numba_cuda/numba/cuda/tests/nrt/test_nrt.py
index 6c77c9e..39e9162 100644
--- a/numba_cuda/numba/cuda/tests/nrt/test_nrt.py
+++ b/numba_cuda/numba/cuda/tests/nrt/test_nrt.py
@@ -40,14 +40,14 @@ def ol_cuda_empty(shape, dtype):
         raise errors.TypingError(msg)
 
 
-@unittest.skip
-class TestNrtRefCt(EnableNRTStatsMixin, TestCase):
+class TestNrtRefCt(EnableNRTStatsMixin, CUDATestCase):
 
     def setUp(self):
         # Clean up any NRT-backed objects hanging in a dead reference cycle
         gc.collect()
         super(TestNrtRefCt, self).setUp()
 
+    @unittest.skip("Stats not yet implemented")
     def test_no_return(self):
         """
         Test issue #1291
```

then the test does get discovered and skipped:

```
$ python -m numba.runtests numba.cuda.tests.nrt.test_nrt -v
test_nrt_launches (numba.cuda.tests.nrt.test_nrt.TestNrtBasic.test_nrt_launches) ... ok
test_nrt_returns_correct (numba.cuda.tests.nrt.test_nrt.TestNrtBasic.test_nrt_returns_correct) ... ok
test_no_return (numba.cuda.tests.nrt.test_nrt.TestNrtRefCt.test_no_return)
Test issue #1291 ... skipped 'Stats not yet implemented'

----------------------------------------------------------------------
Ran 3 tests in 2.921s

OK (skipped=1)
```